### PR TITLE
[MINOR][DOC] revisions for spark sql performance tuning to improve readability and grammar

### PR DESCRIPTION
**Pull request by dwsmith1983**
### What changes were proposed in this pull request?

I made some small grammar fixes related to dependent clause followed but independent clauses, starting a sentence with an introductory phrase, using the plural with when are is present in the sentence, and other small fixes to improve readability.

https://spark.apache.org/docs/latest/sql-performance-tuning.html 

Screenshot 2022-11-04 at 15 24 17

### Why are the changes needed?

These changes improve the readability of the Spark documentation for new users or those studying up.

### Does this PR introduce _any_ user-facing change?

Yes, these changes impact the spark documentation.

### How was this patch tested?

No test were created as these changes were solely in markdown.

### CODE REVIEW

**Comment by dwsmith1983** [comment]
@itholic I was going over another topic and made some updates on sql performance tuning as well. I added a screenshot of the markdown. This how you want it correct?

File: docs/sql-performance-tuning.md [hunk 1]
```diff
@@ -295,7 +294,7 @@ AQE converts sort-merge join to broadcast hash join when the runtime statistics
        <td><code>spark.sql.adaptive.autoBroadcastJoinThreshold</code></td>
        <td>(none)</td>
        <td>
-         Configures the maximum size in bytes for a table that will be broadcast to all worker nodes when performing a join. By setting this value to -1 broadcasting can be disabled. The default value is same with <code>spark.sql.autoBroadcastJoinThreshold</code>. Note that, this config is used only in adaptive framework.
+         Configures the maximum size in bytes for a table that will be broadcast to all worker nodes when performing a join. By setting this value to -1, broadcasting can be disabled. The default value is the same with <code>spark.sql.autoBroadcastJoinThreshold</code>. Note that, this config is used only in adaptive framework.
```
Comment (srowen) [inline docs/sql-performance-tuning.md:297]: the same with -> the same as

**Review by srowen — COMMENTED** [review]
OK, any other related files you want to check while your'e here?

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (dwsmith1983) [reply docs/sql-performance-tuning.md:297]: @srowen thanks for the reply. It just sounded a bit strange. I can convert it back if you like.

**Comment by dwsmith1983** [comment]
> OK, any other related files you want to check while your'e here?

I am doing some studying so not sure what other docs I will read and when.

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (srowen) [reply docs/sql-performance-tuning.md:297]: I'm not suggesting revert; it's a further change. "same with" isn't grammatical (either)

Comment (dwsmith1983) [reply docs/sql-performance-tuning.md]: @srowen done

File: docs/sql-performance-tuning.md [hunk 2]
```diff
@@ -77,8 +77,8 @@ that these options will be deprecated in future release as more optimizations ar
     <td><code>spark.sql.files.openCostInBytes</code></td>
     <td>4194304 (4 MB)</td>
     <td>
-      The estimated cost to open a file, measured by the number of bytes could be scanned in the same
-      time. This is used when putting multiple files into a partition. It is better to over-estimated,
+      The estimated cost to open a file, measured by the number of bytes could that be scanned in the same
```
Comment (srowen) [inline docs/sql-performance-tuning.md:80]: One more - this should be "that could be"

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-performance-tuning.md (hunk 2 above)
Comment (dwsmith1983) [reply docs/sql-performance-tuning.md]: Done

**Comment by dwsmith1983** [comment]
Anyone know why it failed this run? None of code has been touched.

**Comment by srowen** [comment]
Yeah it's not related, looks like an error with the master branch right now. We can wait a bit for someone to fix that
`/home/runner/work/spark/spark/connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectDeduplicateSuite.scala:32: method getSession overrides nothing`

**Comment by dwsmith1983** [comment]
@srowen masters seems to be fine now. I merged mastered into my branch to see if it will fix the tests.

**Comment by dwsmith1983** [comment]
@srowen is the failure related to the bot tagging of connect? There is some connect file that was added after this tag.

**Comment by srowen** [comment]
Weird, the error is on code that isn't in the repo (now). Can you fully rebase and force-push your changes?

**Comment by dwsmith1983** [comment]
@srowen so the streaming test suite passed now so it should be fine now.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by itholic** [comment]
Late LGTM.

**Pull request closed without merging**